### PR TITLE
fix: security vulnerability

### PR DESCRIPTION
**Pull request by l-qing**
## Proposed Changes

* Updated the vulnerable dependency versions
* does not panic when errors occur during delete

**Release Note**

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 
 require (
-	github.com/aws/aws-sdk-go v1.37.1 // indirect
-	github.com/emicklei/go-restful v2.15.0+incompatible // indirect
 	github.com/go-logr/zapr v1.2.2
 	github.com/google/go-cmp v0.5.6
 	github.com/google/go-github/v33 v33.0.0
 	github.com/manifestival/client-go-client v0.5.0
-	github.com/manifestival/manifestival v0.7.1
+	github.com/manifestival/manifestival v0.7.2
```
Comment (l-qing) [inline go.mod:10]: 这是修一个 panic 的故障，给社区提了 PR，还一直没处理。背景详见：
https://github.com/knative/operator/pull/1391

File: go.mod [hunk 2]
```diff
@@ -1,20 +1,17 @@
 module knative.dev/operator
 
-go 1.16
+go 1.17
```
Comment (l-qing) [inline go.mod:3]: 升级依赖后，golang 1.16 编译不通过了。

**Review by danielfbm — APPROVED** [review]

**Comment by l-qing** [comment]
为了不影响 commit，先手动合并了。

**Pull request merged**